duckduckgo/duckduckgo-autofill
# Ema/more fixes

**Reviewer:** @shakyShane 
**Asana:** https://app.asana.com/0/0/1202549120967201/f

## Description
Adds a bunch of further fixes.
- Fields of type=hidden were being ignored on form submission so we were missing several usernames, including on high-profile sites like Amazon.
- On Instagram, a `select` element present on page start triggered the identification of a fake form (just a `div`) which then caused real forms to be misidentified because the previous fake form was a parent. I've changed it so that when a field has an explicit `form` assigned to it we identify the explicit `form` and discard the parent which was obviously a false positive (nested forms are not allowed).
- Remove `type=search` from the entry point selector. This is pretty harmless given that the keyword "search" forces `unknown` but saves us some overhead in that we don't run any scan on these fields.

## Steps to test
- Visit instagram.com and click on login and signup. The forms are correctly identified as a login form (you see keys if you have credentials, otherwise nothing) and a signup form (offers to create a password).
- The hidden field stuff is unit tested.
